src/qdwh.cc

```diff
@@ -0,0 +1,410 @@
+// Copyright (c) 2017-2022, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "slate/slate.hh"
+#include "auxiliary/Debug.hh"
+#include "slate/Matrix.hh"
+#include "internal/internal.hh"
+
+namespace slate {
+
+namespace impl {
```

Comment:
Since this is a higher level driver (similar to gesv and posv), that calls other computational routines not internal routines, we don't actually need to put it in `impl` and template it on target. A few steps to take:
Step 1: delete `namespace impl`
[more steps continued below]